public/app/features/explore/TraceView/components/TraceTimelineViewer/SpanDetail/SpanDetailLinkButtons.tsx

```diff
@@ -206,7 +206,22 @@ const createLinkModel = (
         if (link.onClick) {
           link.onClick?.(event);
         } else {
-          locationService.push(link.href);
+          // TODO: Replace with https://github.com/grafana/grafana/issues/103593
+          // We need to handle absolute and relative URLs correctly because when
+          // there are multiple links we group them into a dropdown and not use
+          // the grafana/ui DataLinkButton component which handles relative and
+          // absolute URLs nicely. A nice solution would be to have a separate
+          // component that handles this for us and not pass the onClick in the
+          // SpanLinkModel when link.href is defined (removing the need of having
+          // if (link.onClick) in here.
+          try {
+            // if it's an absolute URL - open it in a new window
+            new URL(link.href);
+            window.open(link.href, '_blank');
+          } catch (e) {
+            // handle relative URLs by changing current UR:
+            locationService.push(link.href);
+          }
```

Comment:
Can we improve validation here please?

- check if it's an absolute url via regex.test,
- ensure url.protocol is either `http` or `https`,
- `window.open(url, '_blank', 'noopener,noreferrer')`

You could also add error logging if you like so it's easier to debug if user adds an external url that we think is invalid.

Reply:
Updated the handling of absolute/relative URLs. Sanitization is already handled for data links by https://github.com/grafana/grafana/blob/v11.6.0/packages/grafana-data/src/text/sanitize.ts#L98